File: src/main/java/de/floribe2000/warships_java/direct/clans/ClanDetails.java
package de.floribe2000.warships_java.direct.clans;

import de.floribe2000.warships_java.direct.api.ErrorContainer;
import de.floribe2000.warships_java.direct.api.IApiResponse;
import de.floribe2000.warships_java.direct.api.IRequestAction;
import de.floribe2000.warships_java.direct.api.Meta;
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status;

import java.util.List;
import java.util.Map;

public class ClanDetails implements IApiResponse {

    private Status status = Status.ERROR;

    /**
     * Details about errors in case of a failed request.
     * <p>Field is null if no errors occurred during the request!</p>
     */
    private ErrorContainer error = null;

    private Meta meta = null;

    private Map<String, ClanEntry> data = null;

    public Status getStatus() {
        return this.status;
    }

    public ErrorContainer getError() {
        return this.error;
    }

    public Meta getMeta() {
        return this.meta;
    }

    public Map<String, ClanEntry> getData() {
        return this.data;
    }

    public static class ClanEntry {

        private int members_count = 0;

        private String name = null;

        private String creator_name = null;

        private long created_at = 0;

        private String tag = null;

        private long updated_at = 0;

        private String leader_name = null;

        private List<Long> members_ids = null;

        private long creator_id = 0;

        private long clan_id = 0;

        private Map<String, ClanMember> members = null;

        public int getMembers_count() {
            return this.members_count;
        }

        public String getName() {
            return this.name;
        }

        public String getCreator_name() {
            return this.creator_name;
        }

        public long getCreated_at() {
            return this.created_at;
        }

        public String getTag() {
            return this.tag;
        }

        public long getUpdated_at() {
            return this.updated_at;
        }

        public String getLeader_name() {
            return this.leader_name;
        }

        public List<Long> getMembers_ids() {
            return this.members_ids;
        }

        public long getCreator_id() {
            return this.creator_id;
        }

        public long getClan_id() {
            return this.clan_id;
        }

        public Map<String, ClanMember> getMembers() {
            return this.members;
        }

        public String getOld_name() {
            return this.old_name;
        }

        public boolean is_clan_disbanded() {
            return this.is_clan_disbanded;
        }

        public long getRenamed_at() {
            return this.renamed_at;
        }

        public String getOld_tag() {
            return this.old_tag;
        }

        public long getLeader_id() {
            return this.leader_id;
        }

        public String getDescription() {
            return this.description;
        }

        public static class ClanMember {

            private ClanRole role = null;

            private long joined_at = 0;

            private long account_id = 0;

            private String account_name = null;

            public ClanRole getRole() {
                return this.role;
            }

            public long getJoined_at() {
                return this.joined_at;
            }

            public long getAccount_id() {
                return this.account_id;
            }

            public String getAccount_name() {
                return this.account_name;
            }
        }

        private String old_name = null;

        private boolean is_clan_disbanded = false;

        private long renamed_at = 0;

        private String old_tag = null;

        private long leader_id = 0;

        private String description = null;
    }

    @Override
    public String toString() {
        return IRequestAction.GSON.toJson(this);
    }
}

File: src/main/kotlin/de/floribe2000/warships_java/direct/clans/ClanDetails.kt
package de.floribe2000.warships_java.direct.clans

import com.google.gson.annotations.SerializedName
import de.floribe2000.warships_java.direct.api.ErrorContainer
import de.floribe2000.warships_java.direct.api.IApiResponse
import de.floribe2000.warships_java.direct.api.IRequestAction.Companion.GSON
import de.floribe2000.warships_java.direct.api.Meta
import de.floribe2000.warships_java.direct.api.typeDefinitions.Status
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ClanDetails(
    override val status: Status = Status.ERROR,

    /**
     * Details about errors in case of a failed request.
     *
     * Field is null if no errors occurred during the request!
     */
    override val error: ErrorContainer? = null,

    override val meta: Meta = Meta(),

    val data: Map<String, ClanEntry> = mapOf(),

    ) : IApiResponse {

    @Serializable
    data class ClanEntry(
        @SerializedName("members_count")
        @SerialName("members_count")
        val membersCount: Int,

        val name: String,

        @SerializedName("creator_name")
        @SerialName("creator_name")
        val creatorName: String? = null,

        @SerializedName("created_at")
        @SerialName("created_at")
        val createdAt: Long = 0,

        val tag: String,

        @SerializedName("updated_at")
        @SerialName("updated_at")
        val updatedAt: Long = 0,

        @SerializedName("leader_name")
        @SerialName("leader_name")
        val leaderName: String? = null,

        @SerializedName("members_ids")
        @SerialName("members_ids")
        val membersIds: List<Long> = listOf(),

        @SerializedName("creator_id")
        @SerialName("creator_id")
        val creatorId: Long = 0,

        @SerializedName("clan_id")
        @SerialName("clan_id")
        val clanId: Long = 0,

        val members: Map<String, ClanMember> = mapOf(),

        @SerializedName("is_clan_disbanded")
        @SerialName("is_clan_disbanded")
        val isClanDisbanded: Boolean = false,

        @SerializedName("old_name")
        @SerialName("old_name")
        val oldName: String? = null,

        @SerializedName("renamed_at")
        @SerialName("renamed_at")
        val renamedAt: Long? = null,

        @SerializedName("old_tag")
        @SerialName("old_tag")
        val oldTag: String? = null,

        @SerializedName("leader_id")
        @SerialName("leader_id")
        val leaderId: Long = 0,

        val description: String? = null,
    ) {
        @Serializable
        data class ClanMember(
            val role: ClanRole = ClanRole.LINE_OFFICER,

            @SerializedName("joined_at")
            @SerialName("joined_at")
            val joinedAt: Long = 0,

            @SerializedName("account_id")
            @SerialName("account_id")
            val accountId: Long = 0,

            @SerializedName("account_name")
            @SerialName("account_name")
            val accountName: String = "",
        )
    }

    override fun toString(): String {
        return GSON.toJson(this)
    }
}